File: src/com/mrpowergamerbr/loritta/commands/vanilla/fun/PedraPapelTesouraCommand.java
package com.mrpowergamerbr.loritta.commands.vanilla.fun;

import java.io.File;
import java.io.IOException;
import java.util.Arrays;
import java.util.List;
import java.util.Map;

import com.google.common.collect.ImmutableMap;
import com.mrpowergamerbr.loritta.Loritta;
import com.mrpowergamerbr.loritta.commands.CommandBase;
import com.mrpowergamerbr.loritta.commands.CommandCategory;
import com.mrpowergamerbr.loritta.commands.CommandContext;
import com.mrpowergamerbr.loritta.utils.Jankenpon;
import com.mrpowergamerbr.loritta.utils.Jankenpon.JankenponStatus;

public class PedraPapelTesouraCommand extends CommandBase {
	@Override
	public String getLabel() {
		return "ppt";
	}

	@Override
	public String getDescription() {
		return "Jogue Pedra, Papel ou Tesoura! (jankenpon, ou a versão abrasileirada: jokenpô)";
	}

	public String getUsage() {
		return "sua escolha";
	}

	public List<String> getExample() {
		return Arrays.asList("pedra", "papel", "tesoura");
	}

	public Map<String, String> getDetailedUsage() {
		return ImmutableMap.<String, String>builder()
				.put("sua escolha", "Pedra, Papel ou Tesoura")
				.build();
	}
	
	@Override
	public CommandCategory getCategory() {
		return CommandCategory.FUN;
	}

	@Override
	public void run(CommandContext context) {
		if (context.getArgs().length >= 1) {
			String val = context.getArgs()[0];

			Jankenpon janken = Jankenpon.getFromLangString(val.toLowerCase());

			if (janken != null) {
				Jankenpon opponent = Jankenpon.values()[Loritta.getRandom().nextInt(Jankenpon.values().length)];

				JankenponStatus status = janken.getStatus(opponent);

				String fancy = null;
				if (status == JankenponStatus.WIN) {
					fancy = "**Parabéns, você ganhou! :)**";
				}
				if (status == JankenponStatus.LOSE) {
					fancy = "**Que pena... você perdeu, mas o que vale é a intenção! :)**";
				}
				if (status == JankenponStatus.DRAW) {
					fancy = "**Empate! Que tal uma revanche? :)**";
				}
				context.sendMessage(context.getAsMention(true) + "Você escolheu " + janken.getEmoji() + ", eu escolhi " + opponent.getEmoji() + "\n" + fancy);
			} else {
				if (val.equalsIgnoreCase("jesus")) {
					String fancy = "**Empate...? 🤔 🤷**";
					context.sendMessage(context.getAsMention(true) + "Você escolheu 🙇 *JESUS CRISTO*🙇, eu escolhi 🙇 *JESUS CRISTO*🙇\n" + fancy);
				} else if (val.equalsIgnoreCase("velberan")) {
					Jankenpon opponent = Jankenpon.values()[Loritta.getRandom().nextInt(Jankenpon.values().length)];
					
					String fancy = "🕹🕹🕹";
					context.sendMessage(context.getAsMention(true) + "Você escolheu 🕹 *VELBERAN*🕹, eu escolhi " + opponent.getEmoji() + "\n" + fancy);
					try {
						context.sendFile(new File(Loritta.FOLDER + "velberan.gif"), "velberan.gif", " ");
					} catch (IOException e) {
						e.printStackTrace();
					}
				} else {
					String fancy = "**Que pena... você perdeu, dá próxima vez escolha algo que seja válido, ok? :)**";
					context.sendMessage(context.getAsMention(true) + "Você escolheu 💩, eu escolhi 🙇 *JESUS CRISTO*🙇\n" + fancy);
				}
			}
		} else {
			context.explain();
		}
	}
}

File: src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/fun/PedraPapelTesouraCommand.kt
package com.mrpowergamerbr.loritta.commands.vanilla.`fun`

import com.mrpowergamerbr.loritta.Loritta
import com.mrpowergamerbr.loritta.commands.AbstractCommand
import net.perfectdreams.loritta.api.commands.CommandCategory
import com.mrpowergamerbr.loritta.commands.CommandContext
import com.mrpowergamerbr.loritta.utils.Jankenpon
import com.mrpowergamerbr.loritta.utils.Jankenpon.JankenponStatus
import com.mrpowergamerbr.loritta.utils.LoriReply
import com.mrpowergamerbr.loritta.utils.locale.LegacyBaseLocale
import java.io.File
import java.io.IOException

class PedraPapelTesouraCommand : AbstractCommand("jankenpon", listOf("pedrapapeltesoura", "ppt"), CommandCategory.FUN) {
	override fun getDescription(locale: LegacyBaseLocale): String {
		return locale["PPT_DESCRIPTION"]
	}

	override fun getUsage(): String {
		return "sua escolha"
	}

	override fun getExamples(): List<String> {
		return listOf("pedra", "papel", "tesoura")
	}

	override fun getDetailedUsage(): Map<String, String> {
		return mapOf("sua escolha" to "Pedra, Papel ou Tesoura")
	}

	override suspend fun run(context: CommandContext,locale: LegacyBaseLocale) {
		if (context.args.isNotEmpty()) {
			val playerValue = context.args[0]

			val janken = Jankenpon.getFromLangString(playerValue.toLowerCase(), locale)

			if (janken != null) {
				val opponent = Jankenpon.values()[Loritta.RANDOM.nextInt(Jankenpon.values().size)]

				val status = janken.getStatus(opponent)

				var fancy: String? = null
				if (status == JankenponStatus.WIN) {
					fancy = "**${context.legacyLocale["PPT_WIN"]} \uD83D\uDE0A**"
				}
				if (status == JankenponStatus.LOSE) {
					fancy = "**${context.legacyLocale["PPT_LOSE"]} \uD83D\uDE42**"
				}
				if (status == JankenponStatus.DRAW) {
					fancy = "**${context.legacyLocale["PPT_DRAW"]} \uD83D\uDE0A**"
				}
				if (fancy == null) {
					return
				}
				val prefix = when (status) {
					JankenponStatus.WIN -> "\uD83C\uDF89"
					JankenponStatus.DRAW -> "\uD83C\uDFF3"
					JankenponStatus.LOSE -> "\uD83C\uDFF4"
				}
				context.reply(
						LoriReply(message = context.legacyLocale["PPT_CHOSEN", janken.emoji, opponent.emoji], prefix = prefix),
						LoriReply(message = fancy, mentionUser = false)
				)
			} else {
				if (playerValue.equals("jesus", ignoreCase = true)) {
					val fancy = "**${context.legacyLocale["PPT_MAYBE_DRAW"]} 🤔 🤷**"
					val jesus = "🙇 *${context.legacyLocale["PPT_JESUS_CHRIST"]}* 🙇"
					context.reply(
							LoriReply(message = context.legacyLocale["PPT_CHOSEN", jesus, jesus], prefix = "\uD83C\uDFF3"),
							LoriReply(message = fancy, mentionUser = false)
					)
				} else {
					val fancy = "**${context.legacyLocale["PPT_INVALID"]} \uD83D\uDE09**"
					val jesus = "🙇 *${context.legacyLocale["PPT_JESUS_CHRIST"]}* 🙇"
					context.reply(
							LoriReply(message = context.legacyLocale["PPT_CHOSEN", "\uD83D\uDCA9", jesus], prefix = "\uD83C\uDFF4"),
							LoriReply(message = fancy, mentionUser = false)
					)
				}
			}
		} else {
			context.explain()
		}
	}
}